packages/x-date-pickers/src/DesktopDateTimePicker/DesktopDateTimePicker.tsx

```diff
@@ -71,15 +86,17 @@ export const DesktopDateTimePicker = React.forwardRef(function DesktopDateTimePi
       PopperProps={PopperProps}
       PaperProps={PaperProps}
       TransitionComponent={TransitionComponent}
-      clearable={clearable}
-      clearText={clearText}
+      components={components}
```

Comment:
I'm not sure it will be safer. The danger to remove one props is during this PR (I promise I'm careful) 

After this modification, to "accidentally" stop passing prop `component` you need to delete the line `component={component}`

Whereas with `components={other.components}` you don't know if it intentional that `components` is still in `other`

Reply:
I was not thinking just for this PR but in general.
But I'm not sure it's a good idea, just those spread in all direction are quite a mess :grimacing: